iron-fish/ironfish
# Properly catch and surface leveldb corruption errors

## Summary
We were treating all errors as OpenErrors which are thought to be thrown
because the database is locked. It turns out that LevelDB also throws an
OpenError if the DB is corrupt as well. These were being surfaced
specifically as locked errors which are not correct.

## Testing Plan
*Node already in use test*
1. Run a node
2. Run another node to see database already being used error

*Database corrupt test*
1. Delete a random file in your leveldb directory
2. Run your node

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```
